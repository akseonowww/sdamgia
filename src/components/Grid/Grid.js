import React from 'react';
import cx from 'classnames';

import './Grid.scss';

const Grid = ({ className, children }) => (
	<div className={cx('Grid', className)}>{children}</div>
);

export default Grid;
