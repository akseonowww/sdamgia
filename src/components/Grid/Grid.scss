.Grid {
	display: grid;
	grid-template-columns: repeat(20, 1fr);
	grid-gap: 0.4rem;
}
