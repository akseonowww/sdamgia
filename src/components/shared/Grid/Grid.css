.Grid {
	display: grid;
	grid-template-columns: repeat(20, 1fr);
	grid-gap: 5px;
}
