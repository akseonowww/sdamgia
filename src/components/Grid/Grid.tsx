import React, { FC, ReactNode } from 'react'
import cx from 'classnames'

import './Grid.scss'

interface IGridProps {
  className: string
  children: ReactNode
}

const Grid: FC<IGridProps> = ({ className, children }) => (
  <div className={cx('Grid', className)}>{children}</div>
)

export default Grid
