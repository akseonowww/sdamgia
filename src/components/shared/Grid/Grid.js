import React from 'react';

import './Grid.css';

const Grid = props => {
	const { className, children } = props;

	return <div className={'Grid' + ' ' + className}>{children}</div>;
};

export default Grid;
