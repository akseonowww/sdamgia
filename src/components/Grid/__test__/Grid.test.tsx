import React from 'react'
import { unmountComponentAtNode, render } from 'react-dom'
import { act } from 'react-dom/test-utils'
import pretty from 'pretty'

import Grid from '../Grid'

describe('Grid', () => {
  let container: HTMLElement | null = null

  beforeEach(() => {
    container = document.createElement('div')
    document.body.appendChild(container)
  })

  afterEach(() => {
    if (container) {
      unmountComponentAtNode(container)
      container.remove()
      container = null
    }
  })

  it('renders with or without a className', () => {
    act(() => {
      render(
        <Grid>
          <div>Auxiliary Block</div>
        </Grid>,
        container
      )
    })
    expect(container && container.textContent).toBe('Auxiliary Block')

    act(() => {
      render(
        <Grid className="ConstructorForm">
          <div>Constructor Form</div>
        </Grid>,
        container
      )
    })
    expect(container && container.textContent).toBe('Constructor Form')
  })

  it('should render a content', () => {
    act(() => {
      render(
        <Grid>
          <div>Auxiliary Block</div>
        </Grid>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <Grid className="ConstructorForm">
          <div>Constructor Form</div>
        </Grid>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()

    act(() => {
      render(
        <Grid className="ConstructorForm">
          <div>Constructor Form</div>
          <div>Constructor Buttons</div>
        </Grid>,
        container
      )
    })
    if (container) expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
