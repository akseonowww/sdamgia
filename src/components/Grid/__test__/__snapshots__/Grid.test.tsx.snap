// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid should render a content 1`] = `
"<div class=\\"Grid\\">
  <div>Auxiliary Block</div>
</div>"
`;

exports[`Grid should render a content 2`] = `
"<div class=\\"Grid ConstructorForm\\">
  <div>Constructor Form</div>
</div>"
`;

exports[`Grid should render a content 3`] = `
"<div class=\\"Grid ConstructorForm\\">
  <div>Constructor Form</div>
  <div>Constructor Buttons</div>
</div>"
`;
